drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1660,6 +1660,8 @@ int usb_dc_ep_disable(const uint8_t ep)
 
 	nrfx_usbd_ep_disable(ep_addr_to_nrfx(ep));
 	ep_ctx->cfg.en = false;
+	ep_ctx->read_pending = false;
+	ep_ctx->write_in_progress = false;
```

Comment:
This makes sense. But shouldn't the endpoint buffer be purged as well, as in `ep_ctx_reset()`?

Reply:
+1

Reply:
You mean to reset buf structure?

```diff
+        ep_ctx->buf.data = ep_ctx->buf.block.data;                                            
+        ep_ctx->buf.curr = ep_ctx->buf.data;                                                  
+        ep_ctx->buf.len  = 0U;
```

Reply:
yes